pulsar-broker/src/main/java/org/apache/pulsar/broker/web/RestEndpointMetricsFilter.java

```diff
@@ -0,0 +1,106 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.broker.web;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.semconv.SemanticAttributes;
+import java.io.IOException;
+import java.util.List;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.Response;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.pulsar.broker.stats.PulsarBrokerOpenTelemetry;
+import org.glassfish.jersey.server.internal.routing.UriRoutingContext;
+import org.glassfish.jersey.server.model.ResourceMethod;
+import org.glassfish.jersey.uri.UriTemplate;
+
+public class RestEndpointMetricsFilter implements ContainerResponseFilter, ContainerRequestFilter {
+    private static final String REQUEST_START_TIME = "requestStartTime";
+    private static final AttributeKey<String> PATH = SemanticAttributes.URL_PATH;
+    private static final AttributeKey<String> METHOD = SemanticAttributes.HTTP_REQUEST_METHOD;
+    private static final AttributeKey<Long> CODE = SemanticAttributes.HTTP_RESPONSE_STATUS_CODE;
+
+    private final DoubleHistogram latency;
+
+    private RestEndpointMetricsFilter(PulsarBrokerOpenTelemetry openTelemetry) {
+        Meter meter = openTelemetry.getMeter();
+        latency = meter.histogramBuilder("pulsar_broker_rest_endpoint_latency")
+                .setDescription("Latency of REST endpoints in Pulsar broker")
+                .setUnit("ms")
+                .setExplicitBucketBoundariesAdvice(List.of(10D, 20D, 50D, 100D, 200D, 500D, 1000D, 2000D))
+                .build();
+    }
+
+    private static volatile RestEndpointMetricsFilter instance;
+
+    public static synchronized RestEndpointMetricsFilter create(PulsarBrokerOpenTelemetry openTelemetry) {
+        if (instance == null) {
+            instance = new RestEndpointMetricsFilter(openTelemetry);
+        }
+        return instance;
+    }
+
+    @Override
+    public void filter(ContainerRequestContext req, ContainerResponseContext resp) throws IOException {
+        Response.StatusType status = resp.getStatusInfo();
+        int statusCode = status.getStatusCode();
+        Attributes attrs;
+        try {
+            UriRoutingContext info = (UriRoutingContext) req.getUriInfo();
+            attrs = getRequestAttributes(info, statusCode);
+        } catch (Throwable ex) {
+            attrs = Attributes.of(PATH, "UNKNOWN", METHOD, req.getMethod(), CODE, (long) statusCode);
```

Comment:
I don't think it's a good practice. For example OOME will vanish like that.
I think if we don't have any specific exception in mind, no point in guarding this part.